### PR TITLE
Remove globals from ilChatroomInstaller

### PR DESCRIPTION
**Pull request by legionth**
Related to https://www.ilias.de/mantis/view.php?id=22851

### CODE REVIEW

**Comment by legionth** [comment]
Ping @mjansenDatabay

File: Modules/Chatroom/classes/class.ilChatroomInstaller.php [hunk 1]
```diff
@@ -532,12 +534,15 @@ public static function createMissinRoomSettingsForConvertedObjects()
...
-		 * @var $tree      ilTree
-		 * @var $ilDB      ilDBInterface
-		 * @var $rbacadmin ilRbacAdmin
-		 */
-		global $tree, $ilDB, $rbacadmin;
+
+		global $DIC;
+
+		/** @var $tree      ilTree */
+		$tree = $DIC['tree'];
```
Comment (mjansenDatabay) [inline Modules/Chatroom/classes/class.ilChatroomInstaller.php]: `$DIC->repositoryTree()`

File: Modules/Chatroom/classes/class.ilChatroomInstaller.php [hunk 2]
```diff
@@ -532,12 +534,15 @@ public static function createMissinRoomSettingsForConvertedObjects()
...
+		/** @var ilDB $ilDB */
+		$ilDB = $DIC->database();
+		/** @var ilRbacAdmin $rbacadmin */
+		$rbacadmin = $DIC['rbacadmin'];
```
Comment (mjansenDatabay) [inline Modules/Chatroom/classes/class.ilChatroomInstaller.php]: `$DIC->rbac()>admin()`

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

File: Modules/Chatroom/classes/class.ilChatroomInstaller.php [hunk 3]
```diff
@@ -532,12 +534,13 @@ public static function createMissinRoomSettingsForConvertedObjects()
...
-		/**
-		 * @var $tree      ilTree
-		 * @var $ilDB      ilDBInterface
-		 * @var $rbacadmin ilRbacAdmin
-		 */
-		global $tree, $ilDB, $rbacadmin;
+		global $DIC;
+		/** @var $tree      ilTree */
+		$tree = $DIC->repositoryTree();
+		/** @var ilDB $ilDB */
```
Comment (mjansenDatabay) [inline Modules/Chatroom/classes/class.ilChatroomInstaller.php]: ilDBInterface

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**